test/python/test_wrapper.py

```diff
@@ -92,6 +93,28 @@ def test_unregister_non_existent(self):
             qiskit.wrapper.unregister('provider1')
         self.assertEqual(initial_providers, registered_providers())
 
+    @requires_qe_access
+    def test_register_backend_name_conflicts(self, QE_TOKEN, QE_URL,
+                                             hub, group, project):
+        """Test backend name conflicts when registering."""
+
+        qiskit.wrapper.register(QE_TOKEN, QE_URL, hub, group, project,
+                                provider_name='provider1')
+        initial_providers = registered_providers()
+        initial_backends = qiskit.wrapper.available_backends()
+        with self.assertLogs(level=logging.WARNING) as context:
+            qiskit.wrapper.register(QE_TOKEN, QE_URL, hub, group, project,
+                                    provider_name='provider2')
+
+        # Check that a warning has been issued.
+        self.assertEqual(len(context.output), 1)
+        # Check that the provider has been registered.
+        self.assertCountEqual(initial_providers + ['provider2'],
+                              registered_providers())
+        # Check that no new backends have been added.
+        self.assertCountEqual(initial_backends,
```

Comment:
We should also assert that the backend name still refers to the already registered backend:

```python
initial_backends = qiskit.wrapper.available_backends()
original_backend = qiskit.wrapper.get_backend('ibmqx4')
# ...
with self.assertLogs(level=logging.WARNING) as context:
  qiskit.wrapper.register(QE_TOKEN, QE_URL, hub, group, project, provider_name='provider2')
# ...
self.assertEqual(original_backend, qiskit.wrapper.get_backend('ibmqx4'))
```